src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -470,20 +443,17 @@ def _mapLocation(self, fwkJobReport):
                 file.location = self.phedex.getBestNodeName(file.location, self.locLists)
 
 
-    def handleSuccessful(self, jobID, fwkJobReport, fwkJobReportPath = None):
+    def handleJob(self, jobID, fwkJobReport):
         """
-        _handleSuccessful_
+        _handleJob_
 
-        Handle a successful job, parsing the job report and updating the job in
-        WMBS.
-        """
-        wmbsJob = Job(id = jobID)
-        wmbsJob.load()
-        wmbsJob["outcome"] = "success"
-        wmbsJob.getMask()
-        outputID = wmbsJob.loadOutputID()
+        Figure out if a job was successful or not, handle it appropriately
+        (parse FWJR, update WMBS) and return the success status as a boolean
 
-        wmbsJob["fwjr"] = fwkJobReport
+        """
+        jobSuccess = hasattr(fwkJobReport, 'data') \
```

Comment:
BTW, since you are changing this code... how about adding the 2 hasattr() inside this method 
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/FwkJobReport/Report.py#L1010
?